### PR TITLE
Fix for ONNX export for quantized BERT models

### PR DESCRIPTION
**Pull request by anmarques**
1. Remove quantization of identity branch on BERT models.
2. Replace array quantization in NumPy for torch.quantize_per_tensor.

NOTE: This PR does NOT remove quantization of the identity branch for ResNet models. This will require fixes on the model side as well. Will be addressed in a future PR.

### CODE REVIEW

File: src/sparseml/pytorch/sparsification/quantization/quantize_qat_export.py [hunk 1]
```diff
@@ -323,9 +323,21 @@ def _attribute_to_kwarg(attribute: onnx.AttributeProto):
...
+    elif dtype == numpy.int32:
+        tensor_dtype = torch.qint32
+
+    t = torch.Tensor(array).to(torch.float32)
+    if isinstance(scale, numpy.ndarray):
+        scale = scale.item()
+    if isinstance(zero_point, numpy.ndarray):
+        zero_point = zero_point.item()
+
+    q_tensor = torch.quantize_per_tensor(t, scale, zero_point, tensor_dtype)
```
Comment (bfineran) [inline src/sparseml/pytorch/sparsification/quantization/quantize_qat_export.py:339]: nice catch

File: src/sparseml/pytorch/sparsification/quantization/quantize_qat_export.py [hunk 2]
```diff
@@ -323,9 +323,21 @@ def _attribute_to_kwarg(attribute: onnx.AttributeProto):
...
+    return q_tensor.int_repr().numpy()
```
Comment (bfineran) [inline src/sparseml/pytorch/sparsification/quantization/quantize_qat_export.py:340]:
```suggestion
    tensor = torch.Tensor(array).to(torch.float32)
    if isinstance(scale, numpy.ndarray):
        scale = scale.item()
    if isinstance(zero_point, numpy.ndarray):
        zero_point = zero_point.item()

    quant_tensor = torch.quantize_per_tensor(tensor, scale, zero_point, tensor_dtype)
    return quant_tensor.int_repr().numpy()
```

File: src/sparseml/pytorch/sparsification/quantization/quantize_qat_export.py [hunk 3]
```diff
@@ -1569,7 +1585,6 @@ def quantize_torch_qat_export(
     _convert_quantizable_gemm_no_activations(model)
     _quantize_qat_embedding(model)
     quantize_resnet_identity_add_inputs(model)
-    quantized_residual_add_optim(model)
```
Comment (bfineran) [inline src/sparseml/pytorch/sparsification/quantization/quantize_qat_export.py:1572]: let's delete this function if it is no longer used elsewhere in sparseml

**Review by bfineran — COMMENTED** [review]
LGTM pending comments

File: src/sparseml/pytorch/sparsification/quantization/quantize_qat_export.py (hunk 1 above)
Comment (anmarques) [reply src/sparseml/pytorch/sparsification/quantization/quantize_qat_export.py:339]: 🙂

**Review by natuan — DISMISSED** [review]
LGTM, pending Ben's suggestion to delete the unused func

**Review by bfineran — APPROVED** [review]

File: src/sparseml/pytorch/sparsification/quantization/quantize_qat_export.py (hunk 3 above)
Comment (anmarques) [reply src/sparseml/pytorch/sparsification/quantization/quantize_qat_export.py:1572]: Done

**Review by natuan — APPROVED** [review]

**Pull request merged**